src/liballoc/rc.rs

```diff
@@ -312,6 +312,43 @@ impl<T> Rc<T> {
         }
     }
 
+    /// Construct a new Rc with uninitialized contents.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(new_uninit)]
+    /// #![feature(get_mut_unchecked)]
+    ///
+    /// use std::rc::Rc;
+    ///
+    /// let mut five = Rc::<u32>::new_uninit();
+    ///
+    /// let five = unsafe {
+    ///     // Deferred initialization:
+    ///     Rc::get_mut_unchecked(&mut five).as_mut_ptr().write(5);
+    ///
+    ///     five.assume_init()
+    /// };
+    ///
+    /// assert_eq!(*five, 5)
+    /// ```
+    #[unstable(feature = "new_uninit", issue = "63291")]
+    pub fn new_uninit() -> Rc<mem::MaybeUninit<T>> {
+        let layout = Layout::new::<RcBox<mem::MaybeUninit<T>>>();
+        unsafe {
+            let mut ptr = Global.alloc(layout)
+                .unwrap_or_else(|_| handle_alloc_error(layout))
+                .cast::<RcBox<mem::MaybeUninit<T>>>();
+            ptr::write(&mut ptr.as_mut().strong, Cell::new(1));
+            ptr::write(&mut ptr.as_mut().weak, Cell::new(1));
+            Rc {
```

Comment:
This could reuse `Rc::from_inner` (https://github.com/rust-lang/rust/pull/61953/files#diff-da456bd3af6d94a9693e625ff7303113R291) -- applies throughout the PR and also applies to `Arc`.

Reply:
Done, as for other suggestions.